homeassistant/components/simplepush/notify.py

```diff
@@ -44,45 +54,120 @@ async def async_get_service(
+            if isinstance(actions_data, list) and actions_data:
+                actions = []
+                for action in actions_data:
+                    if "action" in action and "url" in action:
+                        actions.append({"name": action["action"], "url": action["url"]})
+                    elif "action" in action:
+                        actions.append(action["action"])
+                        if "id" in action:
+                            action_ids.update({action["action"]: action["id"]})
+
+                try:
+                    feedback_action_timeout = int(
+                        data.get(ATTR_FEEDBACK_ACTION_TIMEOUT)
+                    )
+                except (ValueError, TypeError):
+                    feedback_action_timeout = 60
+
+            attachments_data = data.get(ATTR_ATTACHMENTS)
+            if isinstance(attachments_data, list) and attachments_data:
+                attachments = []
+                for attachment in attachments_data:
+                    if "attachment" in attachment and "thumbnail" in attachment:
+                        attachments.append(
+                            {
+                                "video": attachment["attachment"],
+                                "thumbnail": attachment["thumbnail"],
+                            }
+                        )
+                    elif "attachment" in attachment:
+                        attachments.append(attachment["attachment"])
+
         # use event from config until YAML config is removed
         event = event or self._event
 
+        def feedback_action_callback(
+            action_selected, action_selected_at, action_delivered_at, feedback_id
+        ):
+            payload = {
+                "action_selected": action_selected,
+                "action_selected_at": action_selected_at,
+                "action_delivered_at": action_delivered_at,
+                "feedback_id": feedback_id,
+            }
+
+            if action_selected in action_ids:
+                payload.update({"id": action_ids[action_selected]})
+
+            self.hass.bus.async_fire(EVENT_ACTION_TRIGGERED, payload)
+
         try:
             if self._password:
-                send(
-                    key=self._device_key,
-                    password=self._password,
-                    salt=self._salt,
-                    title=title,
-                    message=message,
-                    event=event,
+                self.hass.async_create_task(
+                    async_send(
+                        key=self._device_key,
+                        password=self._password,
+                        salt=self._salt,
+                        title=title,
+                        message=message,
+                        actions=actions,
+                        attachments=attachments,
+                        event=event,
+                        feedback_callback=feedback_action_callback,
```

Comment:
The callback will potentially make a request every second for one minute. That doesn't seem prudent.

https://github.com/simplepush/simplepush-python/blob/52f4cf2b072f42a1dde94698f5049ce122dddfb8/simplepush/__init__.py#L258-L286

`n` is never increased in the loop either.

Reply:
We're not some playground. We're real people spending real time maintaining Home Assistant and reviewing PRs. If you are making something that is only used by yourself and it's full of hacks… it doesn't belong in HA. I suggest to make it available as a custom component that people can install to HACS. There are 900 PRs open across Home Assistant, all intended for a bigger audience.